ADAL/src/cache/ADTokenCache.m

```diff
@@ -69,6 +69,45 @@ - (void)dealloc
     SAFE_ARC_SUPER_DEALLOC();
 }
 
+- (id)cacheObjectForKey:(id<NSCopying>)key
+{
+    id object = nil;
+    @synchronized(self)
+    {
+        object = _cache[key];
+    }
+    return object;
+}
+
+- (void)setCache:(NSMutableDictionary *)newCache
+{
+    @synchronized(self)
+    {
+        SAFE_ARC_RELEASE(_cache);
+        _cache = newCache;
+        SAFE_ARC_RETAIN(_cache);
+    }
+}
+
+- (NSMutableDictionary *)cacheCopy
+{
+    NSMutableDictionary *copy = nil;
+    @synchronized(self)
+    {
+        copy = [_cache mutableCopy];
+    }
+    SAFE_ARC_AUTORELEASE(copy);
```

Comment:
Wait... "starts with" I still don't like it as it's unnecessarily confusing (kind of like Apple's slight differences between the CF and ObjC create rules...)